crypto/doc/nrf_cc310_mbedcrypto.rst

```diff
@@ -186,8 +177,10 @@ An alternative to allocating this on the heap is to provide a reference to a sta
 Usage restrictions
 ------------------
 
-On the nRF9160 SiP, the nrf_cc3xx_mbedcrypto library is restricted to only work in secure processing environment.
-The library uses mutexes to ensure single usage of hardware modules.
+The library can not be used in the non-secure domain of an application that uses ARM TrustZone.
+
+The hardware can only process one request at a time.
+Therefore, this library uses mutexes to ensure that multiple threads can use it in a thread-safe way.
```

Comment:
(My last comment for the day - the rest looks good!)
My colleagues and I were wondering what the term "thread-safe" means here? In other words, is it that some code that implemented makes it "thread-safe"? Or as my colleague put it, we are wondering: "_is it possible to rewrite where we can say "thread-safe code" or does 'way' make more sense? I'm trying to understand if something is made thread-safe, or if the action makes it thread-safe._" 

We are wondering this so we can better decide if this should be one word, two words, or hyphenated for this case and in the future.

Reply:
I'm not sure if my usage of the term was strictly correct, so I re-wrote it to definitely be correct.

thread-safe is a property that an API or, in this case a library has.

We added mutexes to the library to give it this property.

I tried to make it easier for the user by redundantly explaining what was now made safe, but I see this caused confusion.

https://stackoverflow.com/questions/261683/what-is-the-meaning-of-the-term-thread-safe